org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/MavenCentralIdentifier.java

```diff
@@ -85,30 +86,43 @@ private ArtifactKey identifySha1(String sha1, IProgressMonitor monitor) {
 	}
 
 	private ArtifactKey find(String searchUrl, IProgressMonitor monitor) throws IOException, InterruptedException {
-		Duration timeout = Duration.ofSeconds(10);
-		HttpClient client = HttpClient.newBuilder()
-				.connectTimeout(timeout)
-				.proxy(ProxySelector.getDefault())
-	            .version(Version.HTTP_2)
-				.build();
-		HttpRequest httpRequest = HttpRequest.newBuilder()
-				.timeout(timeout)
-		        .uri(URI.create(searchUrl))
-		        .GET()
-		        .build();
-
-		if (monitor == null) {
-			monitor = new NullProgressMonitor();
-		}
-		if (monitor.isCanceled()) {
-			return null;
+		String timeoutStr = System.getProperty("java.lsp.mavensearch.timeout", "10");
+		long seconds;
+		try {
+			seconds = Long.parseLong(timeoutStr);
+		} catch (Exception e) {
```

Comment:
Might as well make this `NumberFormatException`. I don't think `parseLong` throws any other.

Reply:
Fixed.